doc/development_guide/how_tos/custom_checkers.rst

```diff
@@ -218,10 +218,25 @@ Now we can debug our checker!
 .. Note::
 
     ``my_plugin`` refers to a module called ``my_plugin.py``.
-    This module can be made available to pylint by putting this
-    module's parent directory in your ``PYTHONPATH``
-    environment variable or by adding the ``my_plugin.py``
-    file to the ``pylint/checkers`` directory if running from source.
+    The preferred way of making this plugin available to pylint is
+    by installing it as a package. This can be done either from a packaging index like
+    ``PyPI`` or by installing it from a local source such as with ``pip install .``.
+
+    Alternatively, the plugin module can be made available to pylint by
+    putting this module's parent directory in your ``PYTHONPATH``
+    environment variable.
+
+    If your pylint config has an init-hook that modifies
+    ``sys.path`` to include the module's parent directory, this
+    will also work, but only if either:
+
+    * the ``init-hook`` and the ``load-plugins`` list are both
+      defined in a pylintrc file, or...
```

Comment:
```suggestion
      defined in a configuration file, or...
```

I'm slowly nudging people towards using `pyproject.toml` as the `toml` standard is a bit nicer to work with than the `ini` files.